executable/executable_jar.go

```diff
@@ -70,12 +70,62 @@ func LoadExecutableJAR(appPath string) (ExecutableJAR, error) {
 	return ExecutableJAR{}, nil
 }
 
-func findExecutableJAR(appPath string) (string, *properties.Properties, error) {
+func walk(root string, fn filepath.WalkFunc) error {
```

Comment:
This is non-trivial code, so I think we should break this out and put it somewhere else. I think it would make sense for now in an `internal` package. It can then be a public method that's easy to test, but not something that we export as a part of the public API.

I'd also like to see some more test coverage of this. Walking the file system has proven to be a pain and source of issues, even using Go's method. We should try to test some different scenarios like what happens if it hits a file it can't read (no perms), if the file is a symlink, if the initial file doesn't exist, as well as scenarios to make sure it's finding the correct file & the search is operating as expected (i.e. BFS).

Reply:
@dmikusa 

>what happens if it hits a file it can't read (no perms), 

The error is passed to the callback just like with `filepath.Walk()`, then it is up to user what happens.

> if the file is a symlink

`os.Lstat()` is employed so symlink is not followed, just like with  `filepath.Walk()`.

>if the initial file doesn't exist,

Not sure what you mean. Root dir?

Reply:
The point of `walk()` is to behave exactly like `filepath.Walk()` just in a different order.

Reply:
I'll add test for that.

Reply:
> The point of `walk()` is to behave exactly like `filepath.Walk()` just in a different order.

Right, but it's not `filepath.Walk` so we have to test it like the Golang folks test `filepath.Walk` because its now our code. That's why I'm saying I don't think there's enough test coverage here. We need to have thorough coverage of the walk method to make sure it's operating correctly.